lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -863,26 +921,97 @@ void HdVP2Material::_ApplyVP2Fixes(HdMaterialNetwork& outNet, const HdMaterialNe
     outNet.relationships.reserve(numRelationships * 2);
     outNet.primvars.reserve(numNodes);
 
-    // Add passthrough nodes for vector component access.
-    for (const HdMaterialNode& node : tmpNet.nodes) {
-        TfToken primvarToRead;
+    // Add additional nodes necessary for Maya's fragment compiler
+    // to work that are logical predecessors of node.
+    auto addPredecessorNodes = [&](const HdMaterialNode& node) {
+        // If the node is a UsdUVTexture node, verify there is a UsdPrimvarReader_float2 connected
+        // to the st input of it. If not, find the basic st reader and/or create it and connect it
+        if (_IsUsdUVTexture(node)) {
+            // the DrawModeCardsFragment has UsdUVtexture nodes without primvar readers.
+            // Add a primvar reader to each UsdUVTexture which doesn't already have one.
+            if (!stPrimvarReader) {
+                HdMaterialNode stReader;
+                stReader.identifier = UsdImagingTokens->UsdPrimvarReader_float2;
+                stReader.path
+                    = SdfPath(stReader.identifier.GetString() + std::to_string(++nodeCounter));
+                stReader.parameters[_tokens->varname] = _tokens->cardsUv;
+                outNet.nodes.push_back(stReader);
+                stPrimvarReader = &outNet.nodes.back();
+                outNet.primvars.push_back(_tokens->cardsUv);
```

Comment:
This is only correct for the cards draw mode shader. I will have to think of a way to make this safe.